src/maggma/stores/aws.py

```diff
@@ -379,10 +379,17 @@ def write_doc_to_s3(self, doc: Dict, search_keys: List[str]):
                 to_isoformat_ceil_ms(doc[self.last_updated_field])
             )
 
+        # Any underscores are encoded as double dashes in metadata, since keys with 
+        # underscores may be result in the corresponding HTTP header being stripped 
+        # by certain server configurations (e.g. default nginx), leading to:
+        # `botocore.exceptions.ClientError: An error occurred (AccessDenied) when 
+        # calling the PutObject operation: There were headers present in the request 
+        # which were not signed`
+        # Metadata stored in the MongoDB index (self.index) is stored unchanged.
         s3_bucket.put_object(
             Key=self.sub_dir + str(doc[self.key]),
             Body=data,
-            Metadata={k: str(v) for k, v in search_doc.items()},
+            Metadata={k.replace('_', '--'): str(v) for k, v in search_doc.items()},
```

Comment:
Should we do a single `-` instead?

Reply:
Will defer to your judgement here. I suggested the double `--` with the logic that it would be easier to do the reverse operation, assuming that people don't use `--` that often, but perhaps this best avoided since it is not guaranteed.

Reply:
Hmm - I don't feel strongly about it, but I think `--` symbols are a bit confusing. It should be ok with just simplifying to a `_` -> `-` conversion for readability.